contracts/Poll.sol

```diff
@@ -0,0 +1,232 @@
+// SPDX-License-Identifier: MIT
+// DAO - 道可道, 非常道.
+// some of the logic cited from Jackson Ng's ballot.sol and solidity official documentation
+// However, this contract incorporates some theories of DAO to achieve a higher autonomous level
+// Everyone is able to organize a voting event
+
+pragma solidity >=0.8.0;
+
+
+// Preliminary version without time, delegation, blind vote
+contract Poll {
+
+    uint256 constant public registrationPrice = 100000000000000000;
+
+    enum State { VOTING, ENDED }
+    enum Selection { YES, NO, A, B, C, D }
+
+    // struct time{
+    //     uint year;
+    //     uint month;
+    //     uint day;
+    // }
+
+    struct Participant{
+        string voterName;
+        address delegate;
+        address[] represent;
+    }
+
+    struct PollEvent{
+        State state;
+        uint pollId;
+        string name;
+        string description;        
+        uint votingDuration;
+        bool blind; // Show result in real time or in the end 
+        bool aboutDAO; // The prposal is about this dao (tag type)        
+        Selection[] choseFrom;
+        uint totalVote;
+        address[] voted; // An array to store who has already voted 
+        Selection[] votedChoices; // An array to store voter's choices 
+        Selection result;
+    }
+
+    uint public numberOfParticipant;
+    uint private nextPollId = 1;
+    
+    mapping(uint => PollEvent) public polls;
+    mapping(address => Participant) public participants;
+    mapping(string => address) public participantName; 
+
+	constructor() public {
+        numberOfParticipant = 0;
+    }
+
+    event participantRegistered(string name);
+    event participantLookedUp(string name, address _address);
+    event pollCreated(address organizer, string name, uint dur, bool blind, bool aboutDAO);
+    event voteDone(address voter, bool voted);
+
+    modifier newParticipantCheck(string memory name){
+        require(bytes(name).length > 0, "Participant name is empty");
+        require(bytes(participantName[name]).length == 0, "Participant already registered");
+
+        require(msg.value >= registrationPrice, "Asset not enough to register");
+        require(msg.value == registrationPrice, "Amount sent not equal to the registration price");
```

Comment:
What is the registration price for?

Reply:
See the function registerParticipant(). Everyone pays a fee to join the community